src/nunavut/lang/cpp/__init__.py

```diff
@@ -62,6 +82,19 @@ def _handle_stropping_or_encoding_failure(
         # we couldn't help after all. raise the pending error.
         raise pending_error
 
+    def _validate_language_options(self, language_options: typing.Mapping[str, typing.Any]) -> None:
```

Comment:
This functionality belongs in `nunavut.lang.LanguageConfig`. While it's a bad design to code specific configuration rules into the generic LanguageConfig class it is a good stop-gap until we can incorporate yaml schemas of some kind. By putting these warts inside the LanguageConfig abstraction we can minimize the impact of adding schema validation the reset of the code.